app/models/shelter.rb

```diff
@@ -14,4 +14,20 @@ def average_review_rating
       (rating_value.to_f / shelter_reviews.count.to_f)
     end
   end
+
+  def applications
+    applications = []
+    pets.each do |pet|
+      applications << pet.adoption_applications
+    end
+    applications.flatten.uniq.count
+  end
+
+  def approved_applicants
+    approved = []
+    pets.each do |pet|
+      approved << pet.approved_applicant
+    end
+    approved.compact.length
+  end
```

Comment:
This is used in the shelter view to remove the delete button if there are approved applications for pets at the shelter.

Reply:
Great! As we mentioned, with more time, and better knowledge of AR, we could use AR to get the same results.